federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -242,6 +248,24 @@ func (s *ServiceController) Run(workers int, stopCh <-chan struct{}) error {
 	return nil
 }
 
+func (s *ServiceController) init() error {
+	if s.federationName == "" {
+		return fmt.Errorf("ServiceController should not be run without federationName.")
```

Comment:
You will need to add these flags to the controller-manager YAML: https://github.com/kubernetes/kubernetes/blob/master/federation/manifests/federation-controller-manager-deployment.yaml#L22. Otherwise, the e2e tests will start failing as soon as this PR is merged.

You can use the FEDERATION_NAME env var for federation name: https://github.com/kubernetes/kubernetes/blob/6e5faf59b24d326c13d8fb32cf4c2751b78cca97/federation/cluster/common.sh#L96
 You will have to add a similar env var for ZoneName